tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1978,26 +1896,39 @@ Status StorageManager::store_metadata(
   URI metadata_uri;
   RETURN_NOT_OK(metadata->get_uri(uri, &metadata_uri));
 
+  RETURN_NOT_OK(store_data_to_generic_tile(
+      metadata_buff.data(),
+      metadata_buff.size(),
+      metadata_uri,
+      encryption_key));
+  metadata_buff.disown_data();
+  metadata_buff.clear();
```

Comment:
NIT: I don't think that's needed. It will be called in `Buffer` class destructor when `metadata_buf` goes out of scope.